pkg/windows/windows.go

```diff
@@ -591,11 +591,36 @@ func (vm *windows) createDirectories() error {
 // removeDirectories removes all directories created as part of the configuration process
 func (vm *windows) removeDirectories() error {
 	vm.log.Info("removing directories")
+	existingFiles, err := getFilesToTransfer()
+	if err != nil {
+		return errors.Wrapf(err, "error getting list of files transferred to the instance")
+	}
+	for src, dest := range existingFiles {
+		filePath := dest + "\\" + filepath.Base(src.Path)
+		err = wait.Poll(retry.Interval, retry.Timeout, func() (bool, error) {
+			if out, err := vm.Run(rmFileCmd(filePath), true); err != nil {
+				vm.log.V(1).Error(err, "error removing file", "file", filePath, "out", out)
+				return false, nil
+			}
+			return true, nil
+		})
+		if err != nil {
+			return errors.Wrapf(err, "unable to remove file: %s", filePath)
+		}
+	}
 	for _, dir := range RequiredDirectories {
+		// Skip cleanup of k8sDir as it contains the WICD binary and credentials file, required for
+		// running the WICD cleanup command.
+		if dir == K8sDir {
+			continue
+		}
 		if out, err := vm.Run(rmDirCmd(dir), true); err != nil {
 			return fmt.Errorf("unable to remove directory %s, out: %s, err: %s", dir, out, err)
 		}
 	}
+	if out, err := vm.Run(rmDirCmd(K8sDir), true); err != nil {
+		vm.log.V(1).Error(err, "unable to remove directory", "dir", K8sDir, "out", out)
+	}
```

Comment:
I'm trying to understand the reasoning behind why this is the only path being made optional.
```
		// Skip cleanup of k8sDir as it contains the WICD binary and credentials file, required for
		// running the WICD cleanup command.
```
WICD cleanup is ran before this. I'm assuming this is in case this is reconciled again after this point and WICD cleanup cannot be ran. Considering that case is still possible even with this change, have you tried instead checking if the required files for WICD exist [here](https://github.com/openshift/windows-machine-config-operator/blob/master/pkg/windows/windows.go#L414) before calling `RunWICDCleanup`?

As you have already made it so that this function only runs after `RunWICDCleanup`, checking for the WICD files, will make the order of file deletion in this function not matter.

Reply:
What about the corner case where we end up in this path with an out of band WICD binary removal. We will go ahead and attempt to delete rest of the directories without rest of the services being properly cleaned. I would rather we fail on WICD clean up execution.

Reply:
In the Deconfigure case, if an out of band WICD binary removal occurred how would that situation be remediated? WMCO would continuously fail as it is not going to copy those files over anymore, as it is trying to remove the node from the cluster.

Manual intervention would be required, for the user to re-add the instance to the windows-instances ConfigMap, wait for it to be configured, and then remove it from the ConfigMap.

To be clear, I'm suggesting this be approached by making the change within `func (vm *windows) Deconfigure`:

```
	if err := vm.deconfigureWICD(); err != nil {
		return err
	}
	if wicdBinaryExistsOnNode && requiredServiceAccountCredsExistOnNode{
		if err := vm.RunWICDCleanup(apiServerURL, watchNamespace); err != nil {
			return errors.Wrap(err, "Unable to cleanup the Windows instance")
		}
		else {
			vm.log.Warning("Can't clean up service for node...")
		}
	}
```

Reply:
I see your point, we are doing everything here just to ensure we don't end up in a situation where WICD files don't exist on cleanup. If we are able to avoid getting into it without increasing the time required to deconfigure. I am up for it.

Reply:
@sebsoto I still feel maintaining a retry for directory removals would be beneficial, I can use pollImmediate for polling.

Reply:
>In the Deconfigure case, if an out of band WICD binary removal occurred how would that situation be remediated? WMCO would continuously fail as it is not going to copy those files over anymore, as it is trying to remove the node from the cluster.

The situation should not be remediated. 

>To be clear, I'm suggesting this be approached by making the change within func (vm *windows) Deconfigure:

Please don't do this. Lets discuss this as a parking lot item.